src/components/stack/BlockList.js

```diff
@@ -70,22 +72,89 @@ class BlockList extends Component {
     this.fetchBlocks();
   }
 
+  //an array of item id's
+  //int position (index), boolean above
+  handleRearrange(id, above) { //swap stacks[stackFocused].order[top] and stacks[stackFocused].order[bottom]
```

Comment:
Refactor this function into swapBlocks, so it swaps two positions. The caller is responsible for determining which two indices to swap.